integration_test_infrastructure/codebuild/common/codebuild_blueprints/iam_policy_builder.py

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env python
+"""Module responsible of creating IAM roles."""
+from __future__ import print_function
+from os import path
+from awacs.aws import Allow, PolicyDocument, Statement
+from troposphere import AccountId, Join, Partition, Region, iam
+import awacs.logs
+import yaml
+
+BASE_PATH = path.dirname(__file__)
+
+def create_base_policy():
+    """Creates the base policy."""
+    deploy_name_list = ['runway-int-test-']
+    return iam.Policy(
+        PolicyName='base-policy',
+        PolicyDocument=PolicyDocument(
+            Version='2012-10-17',
+            Statement=[
+                Statement(
+                    Action=[
+                        awacs.logs.CreateLogGroup,
+                        awacs.logs.CreateLogStream,
+                        awacs.logs.PutLogEvents
+                    ],
+                    Effect=Allow,
+                    Resource=[
+                        Join(
+                            '',
+                            [
+                                'arn:',
+                                Partition,
+                                ':logs:',
+                                Region,
+                                ':',
+                                AccountId,
+                                ':log-group:/aws/codebuild/'
+                            ] + deploy_name_list + [
+                                '*'
+                            ] + x
+                        ) for x in [[':*'], [':*/*']]
+                    ]
+                )
+            ]
+        )
+    )
+
+
+class IAMPolicyFinder:
+    """Class that finds the corresponding policies for the given integration test."""
+
+    def __init__(self, root=None):
+        """Sets the root path of integration tests."""
+        self.root = root or path.join(
+            BASE_PATH, "..", "..", "..", "..",
+            "integration_tests")
+
+    def file_path(self, test_name):
+        """Gets the policies file path for the given test."""
+        folder = "test_{}".format(test_name.lower())
+        return path.join(self.root, folder, "policies.yaml")
+
+    def find(self, test_name):
+        """Gets the policies for the given integration test."""
+        file_path = path.abspath(self.file_path(test_name))
+        try:
+            policies = []
+            with open(file_path, 'r') as stream:
+                entries = yaml.safe_load(stream)
+                for entry in entries:
+                    policy = iam.Policy(
+                        PolicyName='inline-policy',
+                        PolicyDocument=entry
+                    )
+                    policies.append(policy)
+            return policies
+        except IOError:
+            print("policies.yaml not found for {0} at {1}".format(test_name, file_path))
+            return []
```

Comment:
you can use `path.isfile` to check if the file exists before trying to open so you don't need the `try; except`.

also, use `logging` instead of `print`. `LOGGER` can be initialized at the top of the file with `LOGGER = logging.getLogger(__name__)`

```suggestion
        policies = []
        if path.isfile(file_path):
            with open(file_path, 'r') as stream:
                entries = yaml.safe_load(stream)
                for entry in entries:
                    policy = iam.Policy(
                        PolicyName='inline-policy',
                        PolicyDocument=entry
                    )
                    policies.append(policy)
        else:
            LOGGER.warning('policies.yaml not found for %s at %s', test_name,
                           file_path)
        return policies       
```